### PR TITLE
Query concept [Pending rebase]

### PR DESCRIPTION
**Pull request by tiagojsag**
Query concept

Once the dataset PR is merged, I want to rebase this branch, as that PR modified the same files, and it's too Monday to fix git conflicts after the fact. 

Still, I think the content's here, there isn't much more to add I think, at least for now. We may want to review/add here some more details if it makes sense after we write the actual "how to use" docs, but as far as concept goes, there isn't much more I would add.

### CODE REVIEW

File: source/includes/concept/_dataset.md [hunk 1]
```diff
@@ -69,3 +69,28 @@ Data provided in the form of a JSON document.
 #### Extensible (X) Markup Language (XML)
 
 Data provided in the form of a XML document.
+
+
+## Query
+
+In the previous section, we covered the concept of a RW API dataset which, in simple terms, is a way to tell the RW API that your data exists, and where. While cataloging datasets in a public repository is useful, making that data easily accessible is one of the main goals of the RW API. This is where **queries** come in.
```
Comment (ikas) [inline source/includes/concept/_dataset.md]: I'd add at the end of this section a link to the query endpoint documentation, to keep things referenced: `/index-rw.html#query`.

File: source/includes/concept/_dataset.md [hunk 2]
```diff
@@ -69,3 +69,28 @@ Data provided in the form of a JSON document.
...
+
+
+### Not all SQL constructs are supported
+
+If you've used SQL in the past, you know how powerful (and complex) it can be. Things like nested queries or joins can be hard to use and even more to maintain, even without the added complexity of an environment where multiple data providers coexist. That's why the RW API limits its support to basic SQL syntax, so we can focus on delivering a tool that's simple and easy to use for most users. The sections below will go into more detail on what's supported and what's not, and will help you understand the specifics of what you can achieve with RW API queries.
+
+
+### Some operation will depend on the provider of the dataset you're querying
+
+Our goal is to provide a common querying interface across all datasets, independent of their provider. While we believe we've achieved it in most cases, RW API queries can only be as powerful as the underling provider (and their own APIs) allows them to be. There are cases in which a given SQL construct or function is supported for a given provided, but not for another. In the docs below, you'll find more details on these limitations, per provider.
```
Comment (ikas) [inline source/includes/concept/_dataset.md]: I'd mention here that we provide a more fine-grained description of the support for SQL for different providers on the "Supported SQL syntax reference" - and link `/index-rw.html#supported-sql-syntax-reference`

**Review by ikas — APPROVED** [review]
Some minor adjustments to better reference other content, other than that it looks good to me :+1:

File: source/includes/concept/_dataset.md (hunk 2 above)
Comment (ethanroday) [inline source/includes/concept/_dataset.md]:
```suggestion
Our goal is to provide a common querying interface across all datasets, independent of their provider. While we believe we've achieved it in most cases, RW API queries can only be as powerful as the underling provider (and their own APIs) allows them to be. There are cases in which a given SQL construct or function is supported for a given provider, but not for another. In the docs below, you'll find more details on these limitations, per provider.  
```

**Comment by ethanroday** [comment]
Similar comments to @ikas - I would say anywhere you currently have "in the docs below", that should be replaced with a reference to an actual section.
Otherwise looks great! This is exactly the kind of thing that concept documentation is supposed to accomplish.

**Comment by tiagojsag** [comment]
I agree with both of you, but since both this and the query sections are WIP, perhaps we have a bit of a chicken-and-egg problem. My suggestion is merging this PR, then rebasing the query docs, and adding the links there - if that PR changes the links (as, for example, the dataset PR did) then it's its responsibility to keep them up-to-date. thoughts?

**Comment by tiagojsag** [comment]
We still have a few things to do on the query endpoints docs themselves, so to try and merge this PR first, I am changing the target branch to be the query endpoints docs instead, and I added the links we discussed above.

Let's do a last round of reviews, and if nothing shows up, merge this.

**Pull request merged**